receiver/otlpreceiver/otlp.go

```diff
@@ -224,24 +224,16 @@ func (r *Receiver) startServer(host component.Host) error {
 	err := componenterror.ErrAlreadyStarted
 	r.startServerOnce.Do(func() {
 		err = nil
-		// Register the grpc-gateway on the HTTP server mux
-		c := context.Background()
-		opts := []grpc.DialOption{grpc.WithInsecure()}
-		endpoint := r.ln.Addr().String()
-
-		_, ok := r.ln.(*net.UnixListener)
-		if ok {
-			endpoint = "unix:" + endpoint
-		}
 
-		err = collectortrace.RegisterTraceServiceHandlerFromEndpoint(c, r.gatewayMux, endpoint, opts)
-		if err != nil {
-			return
+		// Register the grpc-gateway on the HTTP server mux
+		if r.traceReceiver != nil {
+			// Cannot return error, impossible to test
+			_ = collectortrace.RegisterTraceServiceHandlerServer(context.Background(), r.gatewayMux, r.traceReceiver)
 		}
 
-		err = collectormetrics.RegisterMetricsServiceHandlerFromEndpoint(c, r.gatewayMux, endpoint, opts)
-		if err != nil {
-			return
+		if r.metricsReceiver != nil {
+			// Cannot return error, impossible to test
+			_ = collectormetrics.RegisterMetricsServiceHandlerServer(context.Background(), r.gatewayMux, r.metricsReceiver)
 		}
 
 		// Start the gRPC and HTTP/JSON (grpc-gateway) servers on the same port.
```

Comment:
This is the `gRPC over HTTP (with the limitation of the trace/span id)` from my comment. I am confused why to disable, also this PR does not change anything related to that, please make a different PR if you think there is a different problem.

Reply:
> I am confused why to disable

In your comment I linked above, you say that we don't want to support JSON yet, but we already do, and there is a known issue with encoding of trace and span IDs.

So to prevent people from using the JSON format until we decide to officially support it, we need to override the default marshaller in gRPC Gateway.

You're right that this is out of scope of this PR, I just made a comment to clarify the desired behaviour since you've touched the gRPC Gateway.

Reply:
Please file an issue to not forget about this. As you said is not in the scope so don't want to focus on other things.

Reply:
#1177